catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/accesscontrol/AccessControlAccessPlugin.java

```diff
@@ -104,6 +109,36 @@ private boolean isOwnerChanged(Metacard prev, Metacard updated) {
     return !isAnyObjectNull(prev, updated) && OWNER_HAS_CHANGED.apply(prev, updated);
   }
 
+  private boolean hasRemoveAccess(Metacard updated) {
+    Attribute attr = updated.getAttribute(Metacard.SECURITY);
+    if (attr == null) {
+      return false;
+    }
+    Map<String, Set<String>> map = (Map<String, Set<String>>) attr.getValue();
+    if (map == null) {
+      return false;
+    }
+    KeyValueCollectionPermission securityPermission =
+        new KeyValueCollectionPermission(CollectionPermission.UPDATE_ACTION, map);
+    return securityPermission
+        .getPermissionList()
+        .stream()
+        .filter(p -> p instanceof KeyValuePermission)
+        .map(p -> (KeyValuePermission) p)
+        .anyMatch(
+            p ->
+                "remove-user-access".equals(p.getKey())
+                    && p.getValues().contains(subjectSupplier.get()));
+  }
+
+  private void addRemoveAccess(Metacard prev, Metacard updated) {
+    Map<String, Set<String>> prevMap =
+        (Map<String, Set<String>>) prev.getAttribute(Metacard.SECURITY).getValue();
+    Map<String, Set<String>> updatedMap =
+        (Map<String, Set<String>>) updated.getAttribute(Metacard.SECURITY).getValue();
+    prevMap.put("remove-user-access", updatedMap.get("remove-user-access"));
```

Comment:
❓ does calling this `put` mutate the `prev` metacard? it's adding the `remove-user-access` into the `prevMap`, but is that updating the metacard itself?

Reply:
Yeah `prevMap` will have the same reference so it will be updated in the metacard obj as well